### DISCUSSION TITLE
How to treat HA cluster in Netbox

### DISCUSSION THREAD

**Discussion by kirillmuravyev** [netbox-community/netbox#12738] [Ideas]
Some network devices (e.g. firewall) supports HA cluster setup which is different to virtual chassis as they do not share the chassis.
Within the cluster devices usually share the same configuration with some exceptions,
Within HA cluster device do not share chassis and interfaces are not stacked like with switch stack.

Examples are Palo Alto, Cisci ASA, Checkpoint firewalls
How do you document HA network device clusters and their modes (Active/Passive; Active/Active) ?

**Comment by mtinberg** [comment]
Nothing too special, a Device record for each device, a Cluster record for each HA group (in our case the cluster names are numeric and pre-date Netbox so I made a Script to allocate the next available not relying on db row id), a custom field / naming convention for pri/stby or pri/act failover modes and FHRP groups for any shared IPs, which for PaloAlto firewalls is usually all​ of the IPs except for the management interface.

You can create all the associations by hand or create custom provisioning scripts to help populate the records with the right associations and fields

—

Mark Tinberg ***@***.***>

Division of Information Technology-Network Services

University of Wisconsin-Madison


________________________________
From: Kirill Muravyev ***@***.***>
Sent: Friday, May 26, 2023 5:09 AM
To: netbox-community/netbox ***@***.***>
Cc: Subscribed ***@***.***>
Subject: [netbox-community/netbox] How to treat HA cluster in Netbox (Discussion #12738)


Some network devices (e.g. firewall) supports HA cluster setup which is different to virtual chassis as they do not share the chassis.
Within the cluster devices usually share the same configuration with some exceptions,
Within HA cluster device do not share chassis and interfaces are not stacked like with switch stack.

Examples are Palo Alto, Cisci ASA, Checkpoint firewalls
How do you document HA network device clusters and their modes (Active/Passive; Active/Active) ?

**Comment by kirillmuravyev** [comment]
Thanks @mtinberg for your feedback!
I will try the same approach!
